src/types.ts

```diff
@@ -213,7 +249,7 @@ export type RequiredSigner = FixlenBuffer<typeof KEY_HASH_LENGTH>
 export type Mint = Multiasset<Int>
 
 // Collateral input
-export type Collateral = {
+export type CollateralInput = {
```

Comment:
At the moment no, I think we introduced separate types in order to simplify things in case that some elements are added to collateral inputs. Do you think we should drop `CollateralInput` and `ReferenceInput` and just use `Input` everywhere? (However, we might still want to keep the separate parsing code because of more specfic error messages.)

Reply:
It's OK to have separate parsing code, just the result of the parsing should be the same. I don't see any forthcoming changes to the Input type representing UTXO "coordinates", and the CDDL explicitly uses the same type.

Reply:
`CollateralInput` and `ReferenceInput` dropped in favour of `TransactionInput`. https://github.com/vacuumlabs/cardano-hw-interop-lib/pull/7/commits/aeeb35bd336447cb1aa1e269d451c2eff0e5d50b